### PR TITLE
#215 reservation logic ListView 215rv4

### PR DESCRIPTION
**Pull request by ebercamargo**
#215 reservation logic ListView 215rv4

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#450](https://github.com/timelesslounge/timelessis/issues/450) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#450](https://github.com/timelesslounge/timelessis/issues/450) is assigned to @vryazanov[/z](https://www.0crat.com/u/vryazanov), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/7e0ba4be-60aa-4708-bbc6-33f967e23d1f); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
Job `gh:timelesslounge/timelessis#450` already assigned to @vryazanov, can't assign to @vladarefiev

File: timeless/reservations/views.py [hunk 1]
```diff
@@ -92,45 +101,55 @@ def get(self, company_id):
...
         id=id
     )
 
 
-@bp.route("/delete", methods=["POST"])
-def delete():
-    flash("Delete not yet implemented")
+@bp.route("/delete/<int:id>", methods=["POST"])
+def delete(id):
+    """ Delete reservation """
```
Comment (vryazanov) [inline timeless/reservations/views.py]: @ebercamargo DeleteView already implemented, see `timeless.views` let's use it.

File: timeless/reservations/views.py [hunk 2]
```diff
@@ -92,45 +101,55 @@ def get(self, company_id):
         return jsonify(reservations_json)
 
 
-@bp.route("/")
-def list_reservations(reservations):
+@bp.route("/create", methods=("GET", "POST"))
+def create():
```
Comment (vryazanov) [inline timeless/reservations/views.py]: @ebercamargo You need to use `CreateView` from `timeless/views.py`

File: tests/test_reservation.py [hunk 3]
```diff
@@ -1,8 +1,10 @@
 from datetime import datetime
 from http import HTTPStatus
 
+import  pytest
 
+@pytest.mark.skip(reason="authentication not implemented for reservations")
```
Comment (vryazanov) [inline tests/test_reservation.py]: @ebercamargo I didn't get it. why did you skip the test?

Comment (ebercamargo) [reply tests/test_reservation.py]: Requested to login for the complete test. See the note.

Comment (vryazanov) [reply tests/test_reservation.py]: @ebercamargo but it's not a problem actually. See https://github.com/timelesslounge/timelessis/blob/master/tests/integration/it_companies_test.py#L21

File: timeless/reservations/views.py (hunk 2 above)
Comment (ebercamargo) [reply timeless/reservations/views.py]: @vryazanov The list was created, according to the top of the file 'view.py', I will be set one :todo in the method, requesting another 30 minutes to create/delete...

File: tests/test_reservation.py (hunk 3 above)
Comment (ebercamargo) [reply tests/test_reservation.py]: I did not know this solution, I'll be adjusting. Thanks for the information.

**Comment by emilianodellacasa** [comment]
@ebercamargo closed for pr #463

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vryazanov[/z](https://www.0crat.com/u/vryazanov), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Job [#450](https://github.com/timelesslounge/timelessis/issues/450) is not in the agenda of @vryazanov[/z](https://www.0crat.com/u/vryazanov), can't inspect

**Comment by 0crat** [comment]
The job [#450](https://github.com/timelesslounge/timelessis/issues/450) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@vryazanov According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by ypshenychka** [comment]
@ebercamargo According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>Messages in a job [always](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html#4-avoid-noiseaddress-your-comments) start with a name of a user they are addressed to.

Please correct your code review comments by indicating an addressee in the beginning.

**Comment by vryazanov** [comment]
> @vryazanov According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
> 
> > The code reviewer found at least three problems in the code.
> > Comments were mostly about design problems, not cosmetic issues.
> 
> Only two issues were found during code review.
> Please confirm that you'll try to find at least three major problems while future reviews.

@ypshenychka I confirm

**Comment by ypshenychka** [comment]
@ebercamargo ping

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @vryazanov[/z](https://www.0crat.com/u/vryazanov)

**Comment by 0crat** [comment]
Quality review completed: +6 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request closed without merging**